### PR TITLE
Use IndexSet instead of HashMap in EdmondKarps

### PR DESCRIPTION
**Pull request by samueltardieu**

### CODE REVIEW

**Comment by samueltardieu** [comment]
Wait for benchmark before merging

File: src/directed/edmonds_karp.rs [hunk 1]
```diff
@@ -51,13 +52,18 @@ where
...
     // work with matrices more easily.
-    let size = vertices.len();
-    let reverse = (0..size)
-        .map(|i| (vertices[i], i))
-        .collect::<HashMap<_, _>>();
-    let mut capacities = EK::new(size, reverse[source], reverse[sink]);
+    let reverse = vertices.iter().collect::<IndexSet<_>>();
+    let mut capacities = EK::new(
+        vertices.len(),
+        reverse.get_index_of(source).unwrap(),
```
Comment (cuviper) [inline src/directed/edmonds_karp.rs:58]: `get_index_of` requires `indexmap 1.4.0` (or later).

**Comment by samueltardieu** [comment]
bors try

**Comment by samueltardieu** [comment]
bors try

**Comment by samueltardieu** [comment]
bors cancel

**Pull request merged**